### PR TITLE
Document extbase type converters for TYPO3 v11LTS

### PR DESCRIPTION
**Pull request by linawolf**
It is already documented in main by https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/2254

releases: 11LTS

### CODE REVIEW

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+:php:`\TYPO3\CMS\Extbase\Property\TypeConverter\AbstractTypeConverter`, which
+already implements this interface.
+
+All type converters **should** have **no internal state**, such that they
+can be used as singletons and multiple times in succession.
+
+The registration and configuration of a type converter is done in the extension's
+:file:`ext_localconf.php`:
+
+..  code-block:: yaml
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst:43]:
```suggestion
..  code-block:: php
```

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    ExtensionUtility::registerTypeConverter(MyDatetimeConverter::class);
+
+..  tip::
+    Starting with TYPO3 v12.0 a type converter is registered in the extension's
+    :file:`Services.yaml`. See
+    :ref:`Type converters in TYPO3 v12 <t3coreapi12:extbase_Type_converters>`.
+
+    To provide compatibility with both TYPO3 v11 and v12 register the type
+    converter in the Services.yaml and keep the call to
+    :php:`ExtensionUtility::registerTypeConverter()` until dropping v11 support.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst:65]: Hmm, I suggest to surround the call with a condition of the TYPO3 version:

    if ((new \TYPO3\CMS\Core\Information\Typo3Version())->getMajorVersion() < 12) 
    {
        ExtensionUtility::registerTypeConverter();
    }
    
This way, the deprecation message is omitted in v12. This makes it also easier later to find the calls to delete when removing v11 support.

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    // Register type converters
+    ExtensionUtility::registerTypeConverter(MyDatetimeConverter::class);
+
+..  tip::
+    Starting with TYPO3 v12.0 a type converter is registered in the extension's
+    :file:`Services.yaml`. See
+    :ref:`Type converters in TYPO3 v12 <t3coreapi12:extbase_Type_converters>`.
+
+    To provide compatibility with both TYPO3 v11 and v12 register the type
+    converter in the Services.yaml and keep the call to
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst:64]:
```suggestion
    converter in the :file:`Services.yaml` and keep the call to
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    declare(strict_types=1);
+
+    use MyVendor\MyExtension\TypeConverter\MyDatetimeConverter;
+    use TYPO3\CMS\Extbase\Utility\ExtensionUtility;
+
+    defined('TYPO3') or die();
+
+    // Register type converters
+    ExtensionUtility::registerTypeConverter(MyDatetimeConverter::class);
+
```
Comment (linawolf) [inline Documentation/ExtensionArchitecture/Extbase/Reference/Controller/TypeConverter.rst:57]:
```suggestion
    // Starting  with TYPO3 v12 extbase type converters are registered in Configuration/Services.yaml
    if ((new \TYPO3\CMS\Core\Information\Typo3Version())->getMajorVersion() < 12) 
    {
        // Register type converters
        ExtensionUtility::registerTypeConverter(MyDatetimeConverter::class);
    }

```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**